### ISSUE TITLE
Inference on big graphs 

### ISSUE THREAD

**Issue by Monketo** [rusty1s/pytorch_geometric#1997]
## ❓ Questions & Help

Hi @rusty1s !
I'm a little bit confused on how to make an inference for big graphs. 
So looking at [reddit.py](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/reddit.py) and [infomax_inductive.py](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/infomax_inductive.py) I see 2 different ways to compute representation for all nodes in a graph. 

**Layer by layer** 

    `def inference(self, x_all):
                  pbar = tqdm(total=x_all.size(0) * self.num_layers)
                  pbar.set_description('Evaluating')

        # Compute representations of nodes layer by layer, using *all*
        # available edges. This leads to faster computation in contrast to
        # immediately computing the final representations of each batch.
        for i in range(self.num_layers):
            xs = []
            for batch_size, n_id, adj in subgraph_loader:
                edge_index, _, size = adj.to(device)
                x = x_all[n_id].to(device)
                x_target = x[:size[1]]
                x = self.convs[i]((x, x_target), edge_index)
                if i != self.num_layers - 1:
                    x = F.relu(x)
                xs.append(x.cpu())

                pbar.update(batch_size)

            x_all = torch.cat(xs, dim=0)

        pbar.close()

        return x_all`

**immediately computing the final representations of each minibatch** 

    `def test():

        model.eval()
        zs = []
        for i, (batch_size, n_id, adjs) in enumerate(test_loader):
            adjs = [adj.to(device) for adj in adjs]
            zs.append(model(x[n_id], adjs)[0])
        z = torch.cat(zs, dim=0)
        train_val_mask = data.train_mask | data.val_mask
        acc = model.test(z[train_val_mask], y[train_val_mask], z[data.test_mask],
                         y[data.test_mask], max_iter=10000)
        return acc
    `

both using  NeighborSampler to iterate through nodes and generate subgraphs. So, from my experiments,  these 2 approaches result in a different node representations.  I guess the difference is that we use features from the whole graph when computing final representations in the **layer by layer**  approach instead of using only minibatch subgraph for the 2nd approach. Do I get this right? Could you please elaborate on this.

**Comment by Monketo** [comment]
Duplicate
